File: API/DTOs/AppointmentDTO.cs
namespace API.DTOs;

public class AppointmentDTO
{
    public int Id { get; set; }
    public int PatientId { get; set; }
    public string PatientName { get; set; }
    public int DoctorId { get; set; }
    public string DoctorName { get; set; }
    public DateTime AppointmentDate { get; set; }
    public string VisitType { get; set; }
    public string Notes { get; set; }
    public string Diagnosis { get; set; }
    public List<PrescriptionDetailDTO> Prescriptions { get; set; } = new();
}

File: ClientApp/src/app/_models/appointmentDTO.ts
import { PrescriptionDetailDTO } from "./prescriptionDetailDTO";

export interface AppointmentDTO {
  id: number;

  patientId: number;
  patientName: string;

  doctorId: number;
  doctorName: string;

  appointmentDate: string;
  visitType: string;
  notes: string;
  diagnosis: string;

  prescriptions: PrescriptionDetailDTO[];
}
